OpenSim/Tests/Wrapping/testWrapCylinder.cpp

```diff
@@ -0,0 +1,314 @@
+ *                                                                            *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may    *
+ * not use this file except in compliance with the License. You may obtain a  *
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0.         *
+ *                                                                            *
+ * Unless required by applicable law or agreed to in writing, software        *
+ * distributed under the License is distributed on an "AS IS" BASIS,          *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   *
+ * See the License for the specific language governing permissions and        *
+ * limitations under the License.                                             *
+ * -------------------------------------------------------------------------- */
+
+#include <OpenSim/OpenSim.h>
+#include <OpenSim/Auxiliary/auxiliaryTestFunctions.h>
+
+#include <string>
+#include <iostream>
+#include <cmath>
+#include <sstream>
+#include <functional>
+#include <memory>
+
+using namespace OpenSim;
+
+namespace {
+
+    // A path segment determined in terms of the start and end point.
+    struct PathSegment final {
+        SimTK::Vec3 start;
+        SimTK::Vec3 end;
+    };
+
+    std::ostream& operator<<(
+        std::ostream& os,
+        const PathSegment& path)
+    {
+        return os <<
+            "PathSegment{start: " << path.start << ", end: " << path.end << "}";
+    }
+
+}
+
+// Section with helpers for evaluating the wrapping result in the upcoming test.
+namespace {
+
+    // Struct for holding the result of the wrapping solution.
+    struct WrapTestResult final {
+
+        static WrapTestResult NoWrap() {
+            WrapTestResult result;
+            result.noWrap = true;
+            return result;
+        }
+
+        // Path segment on cylinder surface.
+        PathSegment path = {{}, {}};
+        // Path segment length.
+        double length = NAN;
+        // True if there is no wrapping (the other fields don't matter).
+        bool noWrap = false;
+    };
+
+    std::ostream& operator<<(
+        std::ostream& os,
+        const WrapTestResult& result)
+    {
+        if (result.noWrap) {
+            return os << "WrapTestResult{noWrap = true}";
+        }
+        return os <<
+            "WrapTestResult{" <<
+            "path: " << result.path << ", " <<
+            "length: " << result.length << "}";
+    }
+
+    // Struct holding the tolerances when asserting the wrapping result.
+    struct WrappingTolerances final {
+        WrappingTolerances() = default;
+
+        WrappingTolerances(double eps):
+            position(eps),
+            length(eps)
+        {}
+
+        double position = 1e-9;
+        double length = 1e-9;
+    };
+
+    bool IsEqual(
+        double lhs,
+        double rhs,
+        double tolerance)
+    {
+        return std::abs(lhs - rhs) < tolerance;
```

Comment:
Maybe `<=` (e.g. is something being off by exactly one tolerance interval permitted)

Reply:
makes sense. To pass tolerance of zero :)